cnxpublishing/views/admin.py

```diff
@@ -90,3 +95,162 @@ def admin_post_publications(request):
                 })
 
     return {'states': states}
+
+
+@view_config(route_name='admin-add-error-banner', request_method='GET',
+             renderer='cnxpublishing.views:templates/error-banner.html',
+             permission='administer')
+def admin_post_error_banner(request):
+    settings = request.registry.settings
+    db_conn_str = settings[config.CONNECTION_STRING]
+
+    banners = []
+    with psycopg2.connect(db_conn_str) as db_conn:
+        with db_conn.cursor() as cursor:
+            cursor.execute("""\
+                SELECT id, service_state_id, starts, ends, priority, message
+                FROM service_state_messages ORDER BY starts DESC;""")
+            for row in cursor.fetchall():
+                banners.append({
+                    'id': row[0],
+                    'service_state_id': row[1],
+                    'starts': str(row[2]),
+                    'ends': str(row[3]),
+                    'priority': row[4],
+                    'message': row[5],
+                })
+    today = datetime.today()
+    tomorrow = today + timedelta(days=1)
+    return {'start_date': today.strftime("%Y-%m-%d"),
+            'start_time': today.strftime("%H:%M"),
+            'end_date': tomorrow.strftime("%Y-%m-%d"),
+            'end_time': tomorrow.strftime("%H:%M"),
+            'banners': banners}
+
+
+def parse_error_args(request):
```

Comment:
I noticed you did this kind of parsing over in cnx-archive. The better thing to do with this kind of data would be to use a [colander](https://docs.pylonsproject.org/projects/colander/en/latest/), as we do in cnx-authoring. This is fine as is. I just wanted you to be aware of this in the off chance you need to do this type of thing again.